### DISCUSSION TITLE
How to make the hydra-ray plugin work with kuberay instead of aws?

### DISCUSSION THREAD

**Discussion by stweigand97** [facebookresearch/hydra#2723] [General]
As outlined in [this issue](https://github.com/facebookresearch/hydra/issues/2709), I am interested in launching workloads with hydra on an on-premise kubernetes cluster.

Since I posted the above issue, I've looked a bit more into the RayAWS hydra-plugin and believe that I could potentially rewrite it to make it compatible with kuberay (i.e. a kubernetes cluster that runs ray). I would be intereted in the opinion of some core hydra developers on what it would take to make that work.

Since the RayAWSLauncher uses the [autoscaler SDK](https://docs.ray.io/en/latest/_modules/ray/autoscaler/sdk/sdk.html) of ray and points to [this ray autoscaler config](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml), I believe, I could do something similar for the [kuberay autoscaler config](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/kuberay/ray-cluster.complete.yaml).

I would love some opinions and pointers before comitting to this idea 😊 

Thanks for the help!

**Comment by odelalleau** [comment]
Just to set expectations, I'm not sure anyone currently working on Hydra has experience with Kubernetes. So you might be on your own on this one! (I may be wrong though)
(btw your "this issue" link just links back to this discussion)

**Reply by stweigand97** [reply]
Thanks! I've fixed the link 😄
